docs/developer-guide/joins/join-streams-and-tables.md

```diff
@@ -259,9 +261,6 @@ Table-table joins are eventually consistent.
       are no guarantees, which can cause missing results or leftRecord-NULL
       results.
 
-Table-table joins can be joined only on their `PRIMARY KEY` field, and one-to-many
```

Comment:
If we're going to have `Foreign-Key Joins` be its own subsection at the same header level as `Table-Table Joins`, then I think we should still have a sentence in this section on table-table joins to clarify that table-table joins join on the primary key of both tables, and that users should see the foreign-key join section below for non-primary-key joins. Also, should we update the header for this section from `Table-Table Joins` to `Primary-Key Table-Table Joins` (and update `Foreign-Key Joins` to `Foreign-Key Table-Table Joins` too)?